### PR TITLE
feat: hot reload

### PR DESCRIPTION
**Pull request by stuartwdouglas**
JVM languages can now used how reload for instant feedback

### CODE REVIEW

File: jvm-runtime/plugin/common/jvmcommon.go [hunk 1]
```diff
@@ -167,6 +172,9 @@ func (s *Service) SyncStubReferences(ctx context.Context, req *connect.Request[l
 // rebuild must include the latest build context id provided by the request or subsequent BuildContextUpdated
 // calls.
 func (s *Service) Build(ctx context.Context, req *connect.Request[langpb.BuildRequest], stream *connect.ServerStream[langpb.BuildEvent]) error {
+	if req.Msg.RebuildAutomatically {
+		return s.handleDevModeRequest(ctx, req, stream)
+	}
 	events := make(chan updateEvent, 32)
```
Comment (matt2e) [inline jvm-runtime/plugin/common/jvmcommon.go:178]: the rest of this function can be gutted I think as it has a lot of logic for receiving new build contexts and watching for file changes, which only happens if `req.Msg.RebuildAutomatically` is true.

Oh wait I'm actually not sure how you are reacting to build context changes?

Comment (stuartwdouglas) [reply jvm-runtime/plugin/common/jvmcommon.go:178]: I am not handling that properly I think, I need to move some of this logic into the hot reload path.

Comment (stuartwdouglas) [reply jvm-runtime/plugin/common/jvmcommon.go]: I think I am handling this properly now

**Comment by stuartwdouglas** [comment]
I need to add some tests around this, but I think they need a different approach to the existing language tests, as most of the logic is handled by Quarkus itself.

**Pull request merged**